client/models.py

```diff
@@ -84,6 +91,37 @@ class Volunteer(Timestampable):
         ("AGRICULTURE", "חקלאות")
     )
 
+    _original_values = {
+        'first_name': None,
+        'last_name': None,
+        'tz_number': None
+    }
+
+    def _update_original_values(self):
+        """
+        Updates the dict used to keep track of changes in certain fields between changes
+        """
+        self._original_values['tz_number'] = self.tz_number
+        self._original_values['first_name'] = self.first_name
+        self._original_values['last_name'] = self.last_name
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._update_original_values()
+
+    def save(self, *args, **kwargs):
+        if self.id is not None:
+            for key, value in self._original_values.items():
+                if getattr(self, key) != value:
+                    # A change occurred in one of the fields listed in the dict - update valid certificates images
+                    for certificate in self.get_active_certificates():
+                        certificate.create_image()
+
+                    break
+
+        super().save(*args, **kwargs)
+        self._update_original_values()
+
```

Comment:
this code might be looks to me too complex and likely to break in future changes...

Reply:
The only way I see this breaking is if someone refactors one of the fields, in which case they'd have to change this also (because it's a refactor). Could you explain how you see this break?

It's pretty generic - the only "hardcoding" you have here is the dict and the _update_original_values() function, but other than that it's generic.

This is the way to keep track of variables changing, because we have 3 it looks a bit more complex.